design/mvp/02_expressions.md

```diff
@@ -79,25 +93,18 @@ let foo;
 ```
 
 The Crochet implementation of this is very similar to the [TC39
-Proposal](https://github.com/tc39/proposal-do-expressions).
+Proposal](https://github.com/tc39/proposal-do-expressions). One notable
+difference is that `let` and `if` can appear last within a block.
 
-If the last statement within the block ends with a semicolon then value of
-the `do` expression is `undefined`. This mimics Rust's behavior with
-`undefined` standing in for `()`.
+The `;` is optional on the last statement in a block. If it's omitted, the
+statement will be parsed as an expression. This means that a bunch of things
+that aren't currently expressions (looping constructs, class declarations, etc)
+are not allowed to come last.
```

Comment:
@dnerdy I'm still not 100% happy with this since there's still a difference in behaviour between the last line in a block having a `;` or not.  I want to maintain the option to drop it since typing `let foo = if (cond) { 5 } else { 10 };` starts getting awkward if we need `;` inside the blocks here.  I added some TODOs below on this could be made better.